easybuild/framework/extensioneasyblock.py

```diff
@@ -84,6 +84,7 @@ def __init__(self, *args, **kwargs):
             # We can't inherit the 'start_dir' value from the parent (which will be set, and will most likely be wrong).
             # It should be specified for the extension specifically, or be empty (so it is auto-derived).
             self.cfg['start_dir'] = self.ext.get('options', {}).get('start_dir', None)
+            self.cfg['postinstallcmds'] = self.ext.get('options', {}).get('postinstallcmds', None)
```

Comment:
Hmm, I'm a bit confused that this is needed...

Why isn't `postinstallcmds` picked up automatically by default?

Reply:
for the same reason that it's needed for `start_dir` I guess.
without this line the global `postinstallcmds` is used instead, but maybe you know a better way?

Reply:
`start_dir` is a bit special, since that's determined "on the fly" during the installation

There may be easyconfigs that rely on `postinstallcmds` being inherited from the parent (for `start_dir` that wouldn't really make sense I think).

That puts us between a rock and a hard place, I guess...

What happens currently if you set a specific `postinstallcmds` value for an extension (X), and also have a top-level `postinstallcmds` (Y)? X should override Y already. If not, that's a bug that we should definitely figure out and fix...

Reply:
>What happens currently if you set a specific postinstallcmds value for an extension (X), and also have a top-level postinstallcmds (Y)?

to summarize: I have an easyconfig with several extensions. one of them has a `postinstallcmds`, the others don't. there is also a global `postinstallcmds`.
* current situation (without this PR and the easyblock PR)
only the global `postinstallcmds` is executed, and only once at the end during the postprocessing step
* using the easyblock PR only
the global `postinstallcmds` is executed at the end **and for each extension that does not have `postinstallcmds`**.
the `postinstallcmds` of the extension is executed for it's own extension only
* using this PR ad the easyblock PR
everything works as intended

>There may be easyconfigs that rely on postinstallcmds being inherited from the parent 

as is shown in the summary, that is not possible for Python extensions. there is currently no other easyblock that uses postinstallcmds, so that should not be an issue?

>X should override Y already. If not, that's a bug that we should definitely figure out and fix...

that's exactly what this PR does, although I admit I don't fully understand how all this complex stuff works

Reply:
Should perhaps add a comment for why postinstallcmds is handled this way.

Without checking the rest of the code for extensions, this does make sense to me.

@boegel any other comments?

Reply:
This is not required, `postinstallcmds` is already correctly picked up when it's specified for a specific extension.

The problem is that `postinstallcmds` for extensions are just ignored (because `post_install_step` is not run for each extension).

That's fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2404, but only for extensions that are installed for `PythonPackage`, we can/should fix that in framework instead.

What's more is that the current approach in https://github.com/easybuilders/easybuild-easyblocks/pull/2404 results in running the `postinstallcmds` *twice* if an easyconfig uses `PythonPackage` as easyblock (so a standalone installation, not an extension).

We can avoid that in framework by only picking up `postinstallcmds` for an extension via `Extension`.